src/Recruiter/Job/Repository.php

```diff
@@ -241,6 +241,181 @@ public function recentHistory($group = null, T\Moment $at = null, array $query =
+            ]
+        ]);
+    }
+
+    public function expiredButStillScheduledJobs()
+    {
+        return $this->map(
+            $this->scheduled->find([ 
+                'scheduled_at' => [
+                    '$lt' => T\MongoDate::from(T\now())
+                ]
+            ])
+        );
+    }
+
+    public function recentJobsWithManyAttempts(T\Moment $lowerLimit)
+    {
+        $archived = $this->map($this->archived->find([
+            'last_execution.ended_at' => [
+                '$gte' => T\MongoDate::from($lowerLimit),
+            ],
+            'attempts' => [
+                '$gt' => 1
+            ]
+        ]));
+        $scheduled = $this->map($this->scheduled->find([
+            'attempts' => [
+                '$gt' => 1
+            ]
+        ]));
+        return array_merge($archived, $scheduled);
+    }
+
+    public function slowRecentJobs(
+        T\Moment $lowerLimit,
+        T\Moment $upperLimit,
+        $secondsToConsiderJobAsSlow=5
+    )
+    {
+        $archivedArray = $this->archived->aggregate([
+            [
+                '$match' => [
+                    'last_execution.ended_at' => [
+                        '$gte' => T\MongoDate::from($lowerLimit),
+                    ],
+                ]
+            ],
+            [
+                '$project' => [
+                    '_id' => '$_id',
+                    'execution_time' => [
+                        '$subtract' => [
+                            '$last_execution.ended_at',
+                            '$last_execution.started_at'
+                        ]
+                    ],
+                    'done' => '$done',
+                    'created_at' => '$created_at',
+                    'locked' => '$locked',
+                    'attempts' => '$attempts',
+                    'group' => '$group',
+                    'workable' => '$workable',
+                    'tags' => '$tags',
+                    'scheduled_at' => '$scheduled_at',
+                    'last_execution' => '$last_execution',
+                    'retry_policy' => '$retry_policy',
+                ]
+            ],
+            [
+                '$match' => [
+                    'execution_time' => [
+                        '$gt' => $secondsToConsiderJobAsSlow*1000 
+                    ]
+                ],
+            ],
+        ])['result'];
+        $archived= [];
+        foreach ($archivedArray as $archivedJob) {
+            $archived[] = Job::import($archivedJob, $this);
+        }
+        $scheduled = $this->map($this->scheduled->find([
+            'scheduled_at' => [
+                '$gte' => T\MongoDate::from($lowerLimit),
+                '$lte' => T\MongoDate::from($upperLimit)
+            ],
+            'last_execution.started_at' => [
+                '$exists' => true,
+            ],
+            'last_execution.ended_at' => [
+                '$exists' => true,
+            ],
+        ]));
```

Comment:
like before, why all scheduled jobs are considered slow?

Reply:
Yes you are right. That's an oversight I am gonna fix thanks